pkg/providers/v1/aws.go

```diff
@@ -5089,6 +5089,13 @@ func IsFargateNode(nodeName string) bool {
 	return strings.HasPrefix(nodeName, fargateNodeNamePrefix)
 }
 
+// extract private ip address from node name
+func nodeNameToIpAddress(nodeName string) string {
+	nodeName = strings.TrimPrefix(nodeName, privateDNSNamePrefix)
+	nodeName = strings.Split(nodeName, ".")[0]
```

Comment:
this breaks for `fargate-192.168.164.88`
see  https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/cloud-provider-aws/471/pull-cloud-provider-aws-test/1562936995968192512 

```
--- FAIL: TestNodeAddressesForFargate (0.00s)
panic: runtime error: index out of range [1] with length 1 [recovered]
	panic: runtime error: index out of range [1] with length 1 
```

Reply:
or rather fargate-ip-192.168.164.88 ? but how could such a name exist. the test might be wrong, I'm not sure.

Reply:
Maybe with dhcp options sets or something if no domain is provided at all. Just guessing.

Reply:
`nodeAddresses, _ := c.NodeAddressesByProviderID(context.TODO(), "aws:///us-west-2c/1abc-2def/fargate-ip-192.168.164.88")` the original test is wrong, it should `fargate-192.168.164.88`